src/test/java/io/vertx/test/core/FileDescriptorLeakRule.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.test.core;
+
+import com.sun.management.UnixOperatingSystemMXBean;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FileDescriptorLeakRule implements TestRule {
+  private static final int BASE_LINE_RUNS = 20;
+  private static final int TEST_RUNS = 20;
+
+  private static final UnixOperatingSystemMXBean os = (UnixOperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
```

Comment:
We cannot assume that `UnixOperatingSystemMXBean` is always present.

We need to instead use the MBeanServer (`ManagementFactory.getPlatformMBeanServer()`)  and invoke the `getOpenFileDescriptorCount` as an MBean attribute via the `MBeanServer`

The rule should do nothing when such attribute does not exist (on Windows).